src/transformStateWithClones.js

```diff
@@ -8,6 +8,32 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      result.push({ ...Object.assign(copyState, action.extraData) });
```

Comment:
[CODE STYLE] - There's no need to use both the spread operator and Object.assign. Choose one method to create a copy and add properties to it. Since you're using the spread operator to create 'copyState', you should use it here as well for consistency.